src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp

```diff
@@ -50,7 +52,13 @@ void __stdcall FiberDetachCallback(void* lpFlsData)
     if (lpFlsData != NULL)
     {
         // The current fiber is the home fiber of a thread, so the thread is shutting down
-        RuntimeThreadShutdown(lpFlsData);
+        // Do not do shutdown for the main thread though.
+        // other threads are terminated before the main one and could leave TLS locked,
```

Comment:
This is not how shutdown works. The shutdown first terminates all threads except the thread that called ExitProcess. ExitProcess can be called by any thread.

Reply:
There is a lot of vague information on the details of process/thread shutdown. It looks like in a completely broad sense it is hard to handle as there is always some crazy scenario such as someone starting new threads in a DLLMain detach handler. 

I think we should strive for handling "common" cases. 
However using NativeAOT in a library and calling ExitProcess from non-main thread would seem reasonable.